### PR TITLE
Fixed eqcd to not run Eq.Eq with no gas

### LINKED ISSUE #237: Standard eq-eq rule is not invertible
It is possible to get to a "point of no return" when automatically applying the equality-equality rule, now that we have relaxed the presuppositions of equality types in the Nuprl style. Therefore, I suppose that the rule which requires well-typedness of the components should not be run by `auto 0`, which we have intended to only apply invertible rules.

@jozefg any thoughts?

### PR DESCRIPTION
**Pull request by jozefg**
This fixes #237 in the obvious way

### CODE REVIEW

File: lib/sml-open-ended [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit 15602b976d830ae7efde16bab71ab7803fcc232b
+Subproject commit 45f7bb33b190de4c449c9da8f844c2cb6869ffb6
```
Comment (jonsterling) [inline lib/sml-open-ended]: is this change intentional?

Comment (jozefg) [inline lib/sml-open-ended]: @jonsterling Oops, I've made more intentional changes. Hang on, I can fix that

**Comment by jonsterling** [comment]
thank you for this!
